Load upcoming events on example page

Piumal1999 opened the pull request:
## Purpose
<!--- Describe the problems, issues, or needs driving this feature/fix and include links to related issues -->
The purpose of this PR is to fix #648

## Goals
<!---  Describe the solutions that this feature/fix will introduce to resolve the problems described above -->
Load upcoming events on example page for testing purposes.

## Approach
<!--- Describe how you are implementing the solutions. Include a link to a Markdown file or Google doc if the feature write-up is too long to paste here. -->
Updated the component to hold a link
Got data in JSON format using Calendar API of sef-data-holder 
Used Mustache.js to render
Loaded upcoming events on a duplicate onelive page

### Screenshots
<!---  Include an animated GIF or screenshot if the change affects the UI.  -->
![8b4d5339-22a4-4c52-b2c7-202e2a98f80d](https://user-images.githubusercontent.com/27498587/78836760-5e658b80-7a10-11ea-9fd1-062ad5815bdd.png)
  
### Preview Link
<!---  This PR will be automatically deployed to surge. -->
<!---  Once you submit the PR, replace "{PR_NUMBER}" with your PR number. -->
https://pr-649-sef-site.surge.sh/onelive/

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Related PRs
<!--- List any other related PRs --> 
N/A

## Test environment
<!--- List all JDK versions, operating systems, databases, and browser/versions on which this feature/fix was tested --> 
Ubuntu 19.04
Google Chrome
JRE: 1.8.0_202-release-1483-b44 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

## Learning
<!--- Describe the research phase and any blog posts, patterns, libraries, or add-ons you used to solve the problem. -->
N/A

Piumal1999 committed:
Add a link to event card component

Piumal1999 committed:
Load upcoming events on onelive page

Piumal1999 committed:
Fix bugs

Gravewalker666 commented on src/main/webapp/onelive/index.html:
```diff
@@ -277,13 +285,75 @@ <h5 class="title">
+    function loadUpcomingEvents() {
+        $.ajax({
+            type: 'GET',
+            url: 'http://localhost:8080/calendar/onelive',
```
localhost:8080?
Isn't this service up and running?

Piumal1999 replied:
Oh yes. I forgot to change it. I'll update

Gravewalker666 commented on src/main/webapp/onelive/index.html:
```diff
@@ -277,13 +285,75 @@ <h5 class="title">
+                        payload.items[i].month = start.toString().substring(4, 7);
+                        payload.items[i].date = start.toString().substring(8, 10);
+                        payload.items[i].startTime = start.toString().substring(16, 21);
+                        payload.items[i].endTime = end.toString().substring(16, 21);
```
Try .toLocaleDateString() and .toLocaleTimeString()

Gravewalker666 replied:
Not necessary but check those methods out.

Piumal1999 replied:
Ok I'll check

Piumal1999 replied:
It works! Thanks :star_struck:

Piumal1999 committed:
Add required changes

Rewrite the function using toLocalDateString method and toLocalTimeString method

Piumal1999 committed:
Change ajax get URL

jayasanka-sack commented on src/main/webapp/onelive/index.html:
```diff
@@ -277,13 +285,75 @@ <h5 class="title">
+                }
+            },
+            error: function () {
+                console.log("Error occurred while loading upcoming events");
```
Console logs should not be in the code.

Piumal1999 replied:
Ok

jayasanka-sack commented on src/main/webapp/examples/components.html:
```diff
@@ -2104,26 +2104,30 @@ <h2 class="mb-5">
-                            <span class="small text-muted mr-4 mb-5">APR, SUN</span>
-                            <span class="mr-5"> 7.00 - 8.00 pm</span>
-                            <span class="font-weight-bold ml-1">Energy harvesting in wearable electronics</span>
+                    <a href="" target="_blank">
+                        <div class="card card-event shadow shadow-lg--hover mb-1">
+                            <div class="card-body">
+                                <span class="h5 mr-1 card-event-date">12</span>
+                                <span class="small text-muted mr-4 text-capitalize">APR, SUN</span>
+                                <span class="mr-5"> 7.00 - 8.00 pm</span>
+                                <span class="font-weight-bold ml-1">Energy harvesting in wearable electronics</span>
+                            </div>
```
Is this a dummy event?

Piumal1999 replied:
Yes
